tensorflow/contrib/image/ops/single_image_random_dot_stereograms_ops.cc

```diff
@@ -0,0 +1,97 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/shape_inference.h"
+
+namespace tensorflow {
+
+using shape_inference::InferenceContext;
+
+REGISTER_OP("SingleImageRandomDotStereograms")
+    .Attr("T: {double,float,int64,int32}")   
+    .Input("depth_values: T")
+    .Output("image: uint8")
+    .Attr("hidden_surface_removal: bool = true")
+    .Attr("convergence_dots_size: int = 8")
+    .Attr("dots_per_inch: int = 72")
+    .Attr("eye_separation: float = 2.5")
+    .Attr("mu: float = .3333")
+    .Attr("normalize: bool = true")
+    .Attr("normalize_max: float = -100.0")
+    .Attr("normalize_min: float = 100.0")
+    .Attr("boarder_level: float = 0.0")
+    .Attr("number_colors: int = 256")    
+    .Attr("generation_mode: string = 'SIRDS'")
+    .Attr("output_image_shape: shape = { dim {size:1024} dim {size: 768} dim {size: 1}}")
+    .Attr("output_data_window: shape = { dim {size:1022} dim {size: 757}}")
+    // TODO(Mazecreator):  Add "SetShapeFN" to reflect proper output shape & Rank 2 error checking on Input
+    // TODO(Mazecreator):  Add better error handling for output shapes / windows settings
+    // .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
+    //   ::tensorflow::shape_inference::ShapeHandle input;
+    //   ::tensorflow::shape_inference::ShapeHandle output;
+    //   TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 2, &input));
+    //   TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 2, &output));
+    //   c->set_output(0,output);
+    //   return Status::OK();
+    // })
+    .Doc(R"doc(
+Output a RandomDotStereogram Tensor of shape "output_image_shape" for export via encode_PNG or encode_JPG OP.
+
+Based upon:
+'http://www.learningace.com/doc/4331582/b6ab058d1e206d68ab60e4e1ead2fe6e/sirds-paper'
+
+Example use which outputs a SIRDS image as picture_out.png:
+img=[[1,2,3,3,2,1],
+     [1,2,3,4,5,2],
+     [1,2,3,4,5,3],
+     [1,2,3,4,5,4],
+     [6,5,4,4,5,5]]
+
+session = tf.InteractiveSession()
+
+sirds = single_image_random_dot_stereograms(img,convergence_dots_size=8,number_colors=256,normalize=True)
+
+out = sirds.eval()
+
+png = tf.image.encode_png(out).eval()
+
+with open('picture_out.png', 'wb') as f:
+    f.write(png)
+
+
+depth_values:           Z values of data to encode into "output_data_window" window, lower further away {0.0 floor(far), 1.0 ceiling(near) after normalization}, must be rank 2

```

Comment:
Can you look at https://www.tensorflow.org/community/documentation, there's a bit of special formatting for the docstring that makes the op documentation look good.